File: 스택_큐/42583/42583.cpp
#include <vector>

using namespace std;

int solution(int bridge_length, int weight, vector<int> truck_weights)
{
   int cnt = 0, l = 0;
   vector<int> v(bridge_length,0);
   for (auto it = truck_weights.begin(); it != truck_weights.end(); ++it)
   {
      while (cnt - v[l] + *it > weight)
      {
         v.push_back(0);
         cnt -= v[l++];
      }
      v.push_back(*it);
      cnt += *it - v[l++];
   }
   return v.size();
}

File: 스택_큐/42583/42583.py
def solution(bridge_length, weight, truck_weights):
    cnt, l = 0, 0
    arr = [0] * bridge_length
    for i in truck_weights:
        while cnt -arr[l] + i > weight:
            arr.append(0)
            cnt -= arr[l]
            l += 1
        arr.append(i)
        cnt += i - arr[l]
        l += 1

    return len(arr)